openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -195,6 +194,17 @@ def get_cohort(user, course_key, assign=True, use_cached=False):
             return None
 
     # Otherwise assign the user a cohort.
+    membership = CohortMembership.objects.create(
```

Comment:
By having this be a create instead of "get the membership, then save", we address the problem that caused this whole PR, where near-simultaneous requests would create a membership, then update it without removing the first CourseUserGroup association.

Since create uses an INSERT when going to SQL, the above scenario will result in the second request trying to commit to database as it finishes (thanks to ATOMIC_REQUSTS) and generate an IntegrityError due to violating the uniqueness constraint on CohortMembership.

This could have been more neatly solved using a get_or_create, but that was adding a significant amount of SQL overhead to simple "get the user's cohort" queries, because the default was being calculated each time. So a potential race condition does still exist in which 2 requests will both see that a user has no cohort and try to assign them to one, but now instead of both succeeding and leaving the database in a bad state, the second will fail and subsequent requests will work properly.